frontend/src/pages/connectionTypes/manage/ConnectionTypeDataFieldModal.tsx

```diff
@@ -75,14 +82,13 @@ export const ConnectionTypeDataFieldModal: React.FC<Props> = ({
   const [properties, setProperties] = React.useState<unknown>(field?.properties || {});
   const [isPropertiesValid, setPropertiesValid] = React.useState(true);
 
-  const envVarValidation =
-    !envVar || ENV_VAR_NAME_REGEX.test(envVar) ? ValidatedOptions.default : ValidatedOptions.error;
-  const isValid =
-    !!fieldType &&
-    isPropertiesValid &&
-    !!name &&
-    !!envVar &&
-    envVarValidation === ValidatedOptions.default;
+  const isEnvVarConflict = !!fields?.find(
+    (f) => f !== field && isConnectionTypeDataField(f) && f.envVar === envVar,
+  );
```

Comment:
Do we need to memo-ize this? I don't think it's that computationally expensive, but I think this would still rerun on every key stroke when `name` or `description` states update?

Reply:
Updated with `useMemo`